crates/miden-lib/asm/note_scripts/P2IDE.masm

```diff
@@ -0,0 +1,201 @@
+
+#! Helper procedure to add all assets of a note to an account.
+#!
+#! Inputs:  []
+#! Outputs: []
+proc.add_note_assets_to_account
+    push.0 exec.note::get_assets
+    # => [num_of_assets, 0 = ptr, ...]
+
+    # compute the pointer at which we should stop iterating
+    mul.4 dup.1 add
+    # => [end_ptr, ptr, ...]
+
+    # pad the stack and move the pointer to the top
+    padw movup.5
+    # => [ptr, 0, 0, 0, 0, end_ptr, ...]
+
+    # compute the loop latch
+    dup dup.6 neq
+    # => [latch, ptr, 0, 0, 0, 0, end_ptr, ...]
+
+    while.true
+        # => [ptr, 0, 0, 0, 0, end_ptr, ...]
+
+        # save the pointer so that we can use it later
+        dup movdn.5
+        # => [ptr, 0, 0, 0, 0, ptr, end_ptr, ...]
+
+        # load the asset
+        mem_loadw
+        # => [ASSET, ptr, end_ptr, ...]
+
+        # pad the stack before call
+        padw swapw padw padw swapdw
+        # => [ASSET, pad(12), ptr, end_ptr, ...]
+
+        # add asset to the account
+        call.wallet::receive_asset
+        # => [pad(16), ptr, end_ptr, ...]
+
+        # clean the stack after call
+        dropw dropw dropw
+        # => [0, 0, 0, 0, ptr, end_ptr, ...]
+
+        # increment the pointer and compare it to the end_ptr
+        movup.4 add.4 dup dup.6 neq
+        # => [latch, ptr+4, ASSET, end_ptr, ...]
+    end
+
+    # clear the stack
+    drop dropw drop
+end
+
+#! Helper procedure to check if the P2IDE note is unlocked.
+#!
+#! Inputs:  [timelock_block_height, reclaim_block_height, target_account_id_prefix, target_account_id_suffix]
+#! Outputs: [reclaim_block_height, target_account_id_prefix, target_account_id_suffix]
+proc.verify_unlocked
+    # get current block number and store in memory for later
+    exec.tx::get_block_number dup mem_store.CURRENT_BLOCK_HEIGHT_MEM_ADDR
+    # => [current_block_height, timelock_block_height, reclaim_block_height, target_account_id_prefix, target_account_id_suffix]
+
+    # check timelock_block_height <= current_block_height
+    lte
+    # => [is_unlockable, reclaim_block_height, target_account_id_prefix, target_account_id_suffix]
+
+    assert.err=ERR_P2IDE_TIMELOCK_HEIGHT_NOT_REACHED
+end
+
+#! Helper procedure which adds the note assets to the sender account.
+#!
+#! Checks if P2IDE reclaim is enabled and if true, if reclaim height has been reached.
+#!
+#! Inputs:  [account_id_prefix, account_id_suffix, reclaim_block_height]
+#! Outputs: []
+#!
+#! Panics if:
+#! - the reclaim of the current note is disabled.
+#! - the reclaim block height is not reached yet.
+#! - the account attempting to reclaim the note is not the sender account.
+proc.consume_by_sender
+    # check that the reclaim of the current note is enabled
+    movup.2 dup neq.0 assert.err=ERR_P2IDE_RECLAIM_DISABLED
+    # => [reclaim_block_height, account_id_prefix, account_id_suffix]
+
+    # now check that sender is allowed to reclaim, current block >= reclaim block height
```

Comment:
```suggestion
    # now check that sender is allowed to reclaim, reclaim block height <= current block height
```

Nit: Since we use `lte`, I would match the expression in the commment to that.